File: 1436-destination-city/1436-destination-city.java
class Solution {
    public String destCity(List<List<String>> p) {
        
        HashSet<String> m=new HashSet<>();
        
        for( int i=0;i<p.size();i++){
            m.add(p.get(i).get(0));
        }
        
        for( int i=0;i<p.size();i++){
            if( !m.contains( p.get(i).get(1) ) ) return p.get(i).get(1);
        }
        
        return "";
        
    }
}

File: 1436-destination-city/1436-destination-city.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& p) {
        unordered_set< string> s;
        for( int i=0; i<p.size();i++)  s.insert(p[i][0]);
        
        for( int i=0;i<p.size() ;i++)  if( !s.count(p[i][1])) return p[i][1];
        
        return "";
        
    }
};